crates/orderbook/src/order_validation.rs

```diff
@@ -394,6 +394,21 @@ impl OrderValidating for OrderValidator {
             },
         }
 
+        // Orders that are placed and priced outside the market (i.e. buying
+        // more than the market can pay or selling less than the market wants)
+        // get flagged as liquidity orders. The reasoning is that these orders
+        // are not intended to be filled immediately and so need to be treated
+        // slightly differently by the protocol.
+        let is_liquidity_order = match &quote {
+            Some(quote) if is_order_outside_market_price(&order.data, quote) => {
+                let order_uid = order.data.uid(domain_separator, &owner);
+                tracing::debug!(%order_uid, ?owner, "order being flagged as outside market price");
```

Comment:
@josojo - I moved the code a bit and added this log here, so we can query `log: "order being flagged as outside market price"` and get an idea of how often this is happening and for which users and orders.

We can keep a close eye on this in the next few weeks and adjust or remove this if it becomes an issue. WDYT?

Reply:
> But if you are sure about it, its for sure approved

I am not sure about it, it was just my expectation. @marcovc, would Quasimodo be able to do this?

Reply:
Quasimodo does not trade liquidity against other liquidity. There must be at least one user order in a cycle that is used for trading. Does this answer this question?

Reply:
@marcovc Imagine a batch:

User Order: 100 USDC -> 100 DAI
Liquidity Order: 1 WETH -> 0.1 stETH

You could then:
100 USDC -AMM-> 0.1 stETH -LiqO-> WETH -AMM-> 1000 DAI

This would give 900 DAI of surplus to the user.

I guess, in this example Quasimodo should be able to do it (since the user order is part of a cycle).

Settlement. Graphs. Are. Confusing. 😭.

Reply:
Yeah, it really depends on the liquidity graph...

Nick, feel free to merge this PR. Probably the risk is quite low and the advantages outweight the small concerns.

Reply:
Lets keep an eye on how often this happens.